collect/src/main/kotlin/com/certora/collect/TreapMap.kt

```diff
@@ -34,24 +34,27 @@ public sealed interface TreapMap<K, V> : PersistentMap<K, V> {
         merger: (K, V?, V?) -> V?
     ): TreapMap<K, V>
 
-    public fun updateValues(
-        transform: (K, V) -> V?
-    ): TreapMap<K, V>
+    public fun <R : Any> updateValues(
+        transform: (K, V) -> R?
+    ): TreapMap<K, R>
```

Comment:
So this **is** a change in the public API right? If `V` was a nullable type, you could use this function to update the values of the map. But now you can't, `updatedValues` is unusable for nullable types. Maybe that was a bug, as I suspect that we removed entries for which `transform` returned null, but still, it means part of the API is unusable depending on your type parameter.

Reply:
I think I'm favor of this change, as it makes explicit that null values are filtered out, where as previously you could write:

```kotlin
 treapMapOf("foo" to 3, "bar" to null).updateValues { _, v -> v }
```

and have this return a map without "bar" (and no `null` keys) but this wasn't reflected in the return type of  `updateValues`

Reply:
Can we document this behavior a bit more explicitly though?

Reply:
You can definitely write this, before and after this change:

var map = treapMapOf("foo" to 3, "bar" to null).updateValues { _, v -> v }

In both cases the result is a map without the entries that had null values.  With this change, the result will be typed as `TreapMap<String, Int>` instead of `TreapMap<String, Int?>>`.

Reply:
I will add some comments.